packages/dataclass-editor/src/context/FieldContext.tsx

```diff
@@ -0,0 +1,34 @@
+import { createContext, useContext } from 'react';
+import type { DataClassField, EntityClassField } from '../data/dataclass';
+
+type FieldContext = {
+  field: DataClassField;
+  setField: (field: DataClassField) => void;
+};
+
+const fieldContext = createContext<FieldContext>({
+  field: {} as DataClassField,
+  setField: () => {}
+});
+
+export const FieldProvider = fieldContext.Provider;
+
+export const useField = () => {
+  return useContext(fieldContext);
+};
+
+type EntityFieldContext = {
+  field: EntityClassField;
+  setField: (field: EntityClassField) => void;
+};
+
+const entityFieldContext = createContext<EntityFieldContext>({
+  field: {} as EntityClassField,
+  setField: () => {}
+});
+
+export const EntityFieldProvider = entityFieldContext.Provider;
+
+export const useEntityField = () => {
+  return useContext(entityFieldContext);
+};
```

Comment:
works of course too, but I think it is enought if you just reuse the FieldContext but make the type "conversion" in the useEntityField hook. Then you wouldn't need to setup another context. Something like:
```tsx
export const useEntityField = () => {
  const {field, setField} = useField();
  if (isEntityField(field)) {
    return {field, setField};
  }
  return undefined;
}
```
But your solution is also fine.

Reply:
But then I will have to check if what is returned from `useEntityField` is undefined or not. This does not seem much different than checking if it is an entity field in the first place or am I missing something?

Reply:
Yes if you want to build a robust save thing than this is most likely the only way that you do some checks.
I mean technically it is also possible that someone uses the useEntityField hook outside of a EntitiyFieldContext... 

And the snippet above I only wanted to show what I mean. I can also be changed that it e.g. do not return undefined, but e.g. an empty object or something the like... Or do a hard casting to the EntityClassField type and return anyways the field object. 

And as said, your solution is not bad :)